content/tboerger/drone-ansible/index.md

```diff
@@ -0,0 +1,84 @@
+---
+version: '1.0'
+date: 2019-02-15T00:00:00+00:00
+title: Ansible
+author: tboerger
+tags: [ infrastructure, ops, ansbile, deploy  ]
+logo: ansible.svg
+repo: drone-plugins/drone-ansible
+image: plugins/ansible
+---
+
+The Ansible plugin can be used to run ansible plays. The below pipeline configuration demonstrates simple usage:
+
+```yaml
+kind: pipeline
+name: default
+
+steps:
+- name: ansible-check
+  image: plugins/ansible:1
+  environment:
+    additional_var:
+      from_secret: additional_var
+    another_var: foo
+  settings:
+    playbook: ansible/playbook.yml
+    galaxy: ansible/requirements.yml
+    inventory: ansible/inventory
+    syntax_check: true
+  when:
+    event:
+    - pull_request
+
+- name: ansible-apply
+  image: plugins/ansible:1
+  environment:
+    additional_var:
+      from_secret: additional_var
+    another_var: foo
+  settings:
+    playbook: ansible/playbook.yml
+    galaxy: ansible/requirements.yml
+    inventory: ansible/inventory
+    private_key:
+      from_secret: ansible_private_key
+    vault_password:
+      from_secret: ansible_vault_password
+  when:
+    event:
+    - push
+    - tag
+
+---
+kind: secret
+
+data:
+  ansible_private_key: ANSIBLE_PRIVATE_KEY
+  ansible_vault_password: ANSIBLE_VAULT_PASSWORD
+  additional_var: ADDITIONAL_VAR
+```
+
+Please use the the secret stores provided by drone, or any external. Do not commit secrets into any public repositories.
+
+Tip: Preferably put your playbooks into a dot-folder, as to reduce interaction with the rest of your repository data, i.e:
+
+    .ansible/
+
+
+# Secret Reference
+
+ANSIBLE_PRIVATE_KEY
```

Comment:
Use parameter names instead of plugin internal environment variables and write it in lower case. E.g this should be:
inventory: Ansible inventory file to use
limit: further limit selected hosts to an additional pattern
vault_password: ....

There are much more possible parameters declared in the main.go of this plugin which should all be documented here :)

Reply:
i will add them, they all come with their respective explanations anyway, so it's pretty straight forward. thanks for actually making me look into the main.go :)
however, i was wondering: multi word flags are usually connected with a '-', but the example is using underscores. is this all interchangeable?

Reply:
While running a plugin, drone automatically converts all plugin parameters to an environment variable with the schema `PLUGIN_<parameter>` so if you set the parameter `vault_password` in your pipeline drone converts it to `PLUGIN_VAULT_PASSWORD`. Drone does not work with the name attribute defined in the go code.

I hope somebody corrects me if I am wrong :D

Reply:
That's right, just take the env variable and drop the plugin_ prefix